types/number.go

```diff
@@ -1,11 +1,82 @@
 package types
 
-import "strconv"
+import (
+	"math"
+	"strconv"
+)
 
 type (
 	Number float64
 )
 
+var ε = math.Nextafter(1, 2) - 1
+
+func NewNumber(a float64) Number {
+	return Number(a)
+}
+
+func (a Number) Value() float64 { return float64(a) }
+
 func (a Number) String() string {
 	return strconv.FormatFloat(float64(a), 'f', -1, 64)
-}
\ No newline at end of file
+}
+
+// https://es5.github.io/#x9.2
+func (a Number) IsTrue() bool {
+	return !a.IsFalse()
+}
+
+func (a Number) IsFalse() bool {
+	return math.IsNaN(float64(a)) ||
+		a == -0.0 ||
+		a == +0.0
+}
+
+// ToBool returns a Boolean according to:
+// https://es5.github.io/#x9.2
+func (a Number) ToBool() Bool {
```

Comment:
I really dislike the idea of all types in a language being booleanables =P.

Reply:
yes, me too.